autometa/taxonomy/get_lca_taxa.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+import os
+import glob
+import logging
+import pyreadr
+
+import pandas as pd
+import numpy as np
+
+from autometa.taxonomy.lca import LCA
+from autometa.taxonomy.ncbi import NCBI, NCBI_DIR
+
+logger = logging.getLogger(__name__)
+
+# TODO: Currently hardcoded. Need to change it. Not sure how.
+
+lca = LCA(dbdir="/media/bigdrive1/Databases/autometa_databases", verbose=True)
+ncbi = NCBI(dirpath="/media/bigdrive1/Databases/autometa_databases", verbose=True)
+
+
+def map_taxid(input_fpath, cog_org_df, output_fpath):
+    """Find the lca of the query and subject seq ID, map it's taxid and taxonomic rank
+
+    Parameters
+    ----------
+    input_fpath : str
+        Path to input dataframe
+    cog_org_df : DataFrame
+        cog-org-20.csv downloaded from the FTP. This maps the assembly accession to the taxid
+    output_fpath : str
+        Path to output directory
```

Comment:
If `output_fpath` is supposed to be the output directory, then why is it `fpath` and not `output_dpath` or `outdir`? Also, should this return the annotated dataframe? There is no way to access this once it is annotated because it is not returned.

```suggestion
def map_taxid(input_fpath: str, cog_org_df: pd.DataFrame, outdir: str) -> pd.DataFrame :
    """Find the lca of the query and subject seq ID, map it's taxid and taxonomic rank

    Parameters
    ----------
    input_fpath : str
        Path to input dataframe
    cog_org_df : DataFrame
        cog-org-20.csv downloaded from the FTP. This maps the assembly accession to the taxid
    outdir : str
        Path to output directory
```


## Second suggestion

Why not just let the use supply the path to write the file? Also, the description to `input_fpath` is unclear and maybe incorrect? You are mentioning a path to an input dataframe. This is a bit confusing because the next argument requires a dataframe. So does this mean filepath or pandas dataframe? Regardless a short descriptor of this data would be helpful.

```suggestion
def map_taxid(input_fpath: str, cog_org_df: pd.DataFrame, out: str) -> pd.DataFrame:
    """Find the lca of the query and subject seq ID, map it's taxid and taxonomic rank

    Parameters
    ----------
    input_fpath : str
        Path to input table
    cog_org_df : DataFrame
        cog-org-20.csv downloaded from the FTP. This maps the assembly accession to the taxid
    out : str
        File path to write dataframe with lca annotations
```